src/Queue.js

```diff
@@ -83,7 +101,12 @@ class Queue {
     // In the case of a PROCESS_BLOCKED interrupt, emit the appropriate scheduler interrupt to the scheduler's interrupt handler
     // In the case of a PROCESS_READY interrupt, emit the appropriate scheduler interrupt to the scheduler's interrupt handler
     emitInterrupt(source, interrupt) {
-
+        let index = this.processes.indexOf(source);
+        let process = this.processes.splice(index, 1);
+        console.log(process);
+        console.log(interrupt);
+        console.log(this.scheduler);
+        this.scheduler.handleInterrupt(this, process, interrupt);
```

Comment:
You started off good with this section.  The complete section should look like this.
```
emitInterrupt(source, interrupt) {
        const sourceIndex = this.processes.indexOf(source);
        this.processes.splice(sourceIndex, 1);

        switch(interrupt) {
            case 'PROCESS_BLOCKED':
                this.scheduler.handleInterrupt(this, source, SchedulerInterrupt.PROCESS_BLOCKED);
                break;
            case 'PROCESS_READY':
                this.scheduler.handleInterrupt(this, source, SchedulerInterrupt.PROCESS_READY);
                break;
            default:
                break;
        }
    }
```